### DISCUSSION TITLE
Both MSC and HID modes together

### DISCUSSION THREAD

**Discussion by cvk786** [hathach/tinyusb#1254] [Ideas]
Hi
On pico or SAMD21 device can i use both MSC and HID modes together? 
Can i have working keyboard+mouse+mass storage for file transfer running on single USB?

**Comment by hathach** [comment]
yeah, that is easy, if you use arduino and adafruit boards, here is the example for that https://github.com/adafruit/Adafruit_TinyUSB_Arduino/tree/master/examples/Composite/mouse_ramdisk

**Reply by cvk786** [reply]
Thank you hathach, it really helpful
